### PR TITLE
Fix mariadb drivers

### PR DESCRIPTION
**Pull request by jamii**
##### Motivation for this change

The mariadb unixODBC drivers fail to load with an unhelpful error message - https://github.com/NixOS/nixpkgs/issues/73258

I haven't been able to pin down what exactly is wrong, but I've found that building against the subrepo rather than against the system libmysqlclient leads to working drivers. 

I've tested this PR against mariadb on NixOS and OS X Mojave.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jamii** [comment]
```
jamie@machine:~/nixpkgs$ nix-shell -p nix-review --run "nix-review wip"
these paths will be fetched (0.03 MiB download, 0.10 MiB unpacked):
  /nix/store/wvrisx65diwaf934clcx7gld33j7b5bg-nix-review-2.0.1
copying path '/nix/store/wvrisx65diwaf934clcx7gld33j7b5bg-nix-review-2.0.1' from 'https://cache.nixos.org'...
$ git fetch --force https://github.com/NixOS/nixpkgs master:refs/nix-review/0
remote: Enumerating objects: 1216, done.
remote: Counting objects: 100% (1216/1216), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 2489 (delta 1212), reused 1214 (delta 1212), pack-reused 1273
Receiving objects: 100% (2489/2489), 1.19 MiB | 2.63 MiB/s, done.
Resolving deltas: 100% (1786/1786), completed with 568 local objects.
From https://github.com/NixOS/nixpkgs
 * [new branch]              master     -> refs/nix-review/0
$ git worktree add /home/jamie/.cache/nix-review/rev-053d2dc590afd23d9ca7f2d354666ff3c6ea628e-dirty/nixpkgs 8960509fc5308dd4ab500f537c30ada256695d7a
Preparing worktree (detached HEAD 8960509fc53)
Updating files: 100% (20343/20343), done.
HEAD is now at 8960509fc53 cargo-make: 0.23.0 -> 0.24.0
$ nix-env -f /home/jamie/.cache/nix-review/rev-053d2dc590afd23d9ca7f2d354666ff3c6ea628e-dirty/nixpkgs -qaP --xml --out-path --show-trace
Applying `nixpkgs` diff...
$ nix-env -f /home/jamie/.cache/nix-review/rev-053d2dc590afd23d9ca7f2d354666ff3c6ea628e-dirty/nixpkgs -qaP --xml --out-path --show-trace --meta
Nothing changed
No packages were successfully build, skip nix-shell
$ git worktree prune
```

**Comment by jamii** [comment]
There don't appear to be any tests.

**Comment by jamii** [comment]
@aanderse

**Comment by aanderse** [comment]
`libmysqlclient` is no longer being referenced so you should remove it. This will also make the backport to `19.09` easier.

File: pkgs/development/libraries/unixODBCDrivers/default.nix [hunk 1]
```diff
@@ -29,35 +29,42 @@
...
-
-    src = fetchurl {
-      url = "https://downloads.mariadb.org/interstitial/connector-odbc-${version}/${pname}-${version}-ga-src.tar.gz";
-      sha256 = "0iibly2mbqijqyq4pzpb6dh40clqhvqrhgnj8knm4bw3nlksd0d5";
+    version = "3.1.4";
+
+    src = fetchFromGitHub {
+      owner = "MariaDB";
+      repo = "mariadb-connector-odbc";
+      rev = "c921b500650eca535010f12e0d83417d2ef915ec";
```
Comment (aanderse) [inline pkgs/development/libraries/unixODBCDrivers/default.nix]: I'm terribly sorry I didn't notice this before... we generally set `rev` to something involving `version` for easy updates. So in this case:

```suggestion
      rev = version;
```

I've tested this on NixOS and after this change I'm happy to merge to `master`. This change deserves a backport as well. Thank you so much for digging into this! Hopefully at some point someone can get to the bottom of our `libmysqlclient` issue, but until that happens the priority is to have a working package :smile:

Thanks again @jamii :tada:

**Review by aanderse — APPROVED** [review]
LGTM :+1:

**Comment by aanderse** [comment]
Thank you very much @jamii! Are you going to open a backport PR?

**Comment by jamii** [comment]
Not soon. I was asked to spend less time working on nix stuff :man_shrugging:

**Comment by aanderse** [comment]
I'll grab it then. Thanks again... and sorry to hear :-(

**Pull request merged**